src/Stratis.Bitcoin.Features.PoA/Voting/VotingManager.cs

```diff
@@ -861,23 +845,12 @@ internal bool Synchronize(ChainedHeader newTip)
                             this.logger.LogInformation(progressString);
                             this.signals.Publish(new FullNodeEvent() { Message = progressString, State = FullNodeState.Initializing.ToString() });
 
-                            this.PollsRepository.SaveCurrentTip(currentTransaction);
-
-                            currentTransaction.Commit();
-                            currentTransaction.Dispose();
-
-                            currentTransaction = this.PollsRepository.GetTransaction();
+                                currentTransaction.Flush();
+                            }
                         }
-                    }
-
-                    // If we ended the synchronization at say block 10100, the current transaction would still be open and
-                    // thus we need to commit and dispose of it.
-                    // If we ended at block 10000, then the current transaction would have been committed and
-                    // disposed and re-opened.
-                    this.PollsRepository.SaveCurrentTip(currentTransaction);
 
-                    currentTransaction.Commit();
-                    currentTransaction.Dispose();
+                        currentTransaction.Commit();
```

Comment:
Note to any other reviewers: at first glance it may appear that there is a `SetTip` missing here when compared to the previous implementation, but it is actually being done inside every call to `ProcessBlock`